core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -100,9 +140,53 @@ public final String path() {
 
     public abstract BytesReference content();
 
-    public abstract String header(String name);
+    /**
+     * Get the value of the header or {@code null} if not found. This method only retrieves the first header value if multiple values are
+     * sent. Use of {@link #getAllHeaderValues(String)} should be preferred
+     */
+    public final String header(String name) {
```

Comment:
can you explain this move? these were previously implemented in the netty request class right?